wearerequired/required-google-analytics
# Support gtag.js dual tagging

Based on https://support.google.com/analytics/answer/11091026?hl=en#gtag-dual-tagging&zippy=%2Cin-this-article

* Add new setting for the measurement ID used by GA 4
* Update script to handle property and/or measurement IDs for config and opt-out

Fixes #14.

### Test instructions

For testing it's recommended to have [Google Analytics Debugger](https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna) installed.

* Checkout branch
* Run `npx wp-env start` (https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/)
* Go to http://localhost:9999/wp-admin/options-reading.php#google-analytics
* Enter IDs
  Examples: `UA-9680182-1` / `G-9VBT77J96N`
* Enable Google Analytics Debugger in current tab
* Open developer tools
* Go to the frontend
* Verify output of Google Analytics Debugger
* Run `window.requiredGADoOptOut();` in the browser console
* Switch to Application tab and delete all `_ga*` cookies
* Reload page and verify that no new cookies are set
